plugins/filter_throttle/throttle.c

```diff
@@ -67,30 +67,29 @@ static bool apply_suffix (double *x, char suffix_char)
 
 void *time_ticker(void *args)
 {
-    struct ticker *t = args;
     struct flb_time ftm;
     long timestamp;
-    struct flb_filter_throttle_ctx *ctx = t->ctx;
+    struct flb_filter_throttle_ctx *ctx = args;
 
-    while (!t->done) {
+    while (1) {
         flb_time_get(&ftm);
         timestamp = flb_time_to_double(&ftm);
-        window_add(t->ctx->hash, timestamp, 0);
+        window_add(ctx->hash, timestamp, 0);
 
-        t->ctx->hash->current_timestamp = timestamp;
+        ctx->hash->current_timestamp = timestamp;
 
-        if (t->ctx->print_status) {
+        if (ctx->print_status) {
             flb_plg_info(ctx->ins,
                          "%ld: limit is %0.2f per %s with window size of %i, "
                          "current rate is: %i per interval",
-                         timestamp, t->ctx->max_rate, t->ctx->slide_interval,
-                         t->ctx->window_size,
-                         t->ctx->hash->total / t->ctx->hash->size);
+                         timestamp, ctx->max_rate, ctx->slide_interval,
+                         ctx->window_size,
+                         ctx->hash->total / ctx->hash->size);
         }
-        sleep(t->seconds);
-    }
 
-    return NULL;
+        // sleep is a cancelable function
```

Comment:
Could you check comment style ?
https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#single-line-comment

`/* sleep is a cancelable function */`

Reply:
will do it tomorrow